### PR TITLE
Issue211

### PR DESCRIPTION
**Pull request by JerryShea**

### CODE REVIEW

File: src/main/java/net/openhft/chronicle/queue/impl/single/StoreAppender.java [hunk 1]
```diff
@@ -653,13 +664,21 @@ void beforeAppend(final Wire wire, final long index) {
...
+    }
+
+    private void rollCycleTo(final int cycle, boolean suppress) {
 
         // only a valid check if the wire was set.
         if (this.cycle == cycle)
             throw new AssertionError();
 
-        store.writeEOF(wire, timeoutMS());
+        if(!suppress) {
```
Comment (nicktindall) [inline src/main/java/net/openhft/chronicle/queue/impl/single/StoreAppender.java]: Could `suppress` have a more descriptive name maybe?

**Review by nicktindall — APPROVED** [review]
Looks sensible to me. Do we need to add documentation to README about lastIndexRead?

It'd be good to get some of that commented code in the test tidied up as well maybe.

File: src/test/java/net/openhft/chronicle/queue/impl/single/StoreAppenderInternalWriteBytesTest.java [hunk 2]
```diff
@@ -28,7 +28,7 @@
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.fail;
 
-@Ignore("Broken https://github.com/ChronicleEnterprise/Chronicle-Queue-Enterprise/issues/211")
+//@Ignore("Broken https://github.com/ChronicleEnterprise/Chronicle-Queue-Enterprise/issues/211")
```
Comment (JerryShea) [inline src/test/java/net/openhft/chronicle/queue/impl/single/StoreAppenderInternalWriteBytesTest.java]: should delete commented-out code

**Review by JerryShea — COMMENTED** [review]
I created the PR so can't approve, but I do approve

File: src/main/java/net/openhft/chronicle/queue/impl/single/StoreAppender.java (hunk 1 above)
Comment (rogersimmons) [reply src/main/java/net/openhft/chronicle/queue/impl/single/StoreAppender.java]: done (renamed suppressEOF)

File: src/test/java/net/openhft/chronicle/queue/impl/single/StoreAppenderInternalWriteBytesTest.java (hunk 2 above)
Comment (rogersimmons) [reply src/test/java/net/openhft/chronicle/queue/impl/single/StoreAppenderInternalWriteBytesTest.java]: done

**Review by rogersimmons — APPROVED** [review]

File: src/main/java/net/openhft/chronicle/queue/impl/single/StoreAppender.java [hunk 3]
```diff
@@ -570,7 +581,7 @@ protected void writeBytesInternal(final long index, @NotNull final BytesStore by
         }
 
         writeBytesInternal(bytes, metadata);
-//        assert !QueueSystemProperties.CHECK_INDEX || checkWritePositionHeaderNumber();
+        //assert !QueueSystemProperties.CHECK_INDEX || checkWritePositionHeaderNumber();
```
Comment (JanStureNielsen) [inline src/main/java/net/openhft/chronicle/queue/impl/single/StoreAppender.java:584]: Remove commented out code?

File: src/main/java/net/openhft/chronicle/queue/impl/single/StoreTailer.java [hunk 4]
```diff
@@ -220,6 +222,7 @@ DocumentContext readingDocument0(final boolean includeMetaData) {
                 context.setStart(bytes.readPosition() - 4);
                 readingDocumentFound = true;
                 address = bytes.addressForRead(bytes.readPosition(), 4);
+                this.lastReadIndex = this.index();
 //                Jvm.optionalSafepoint();
```
Comment (JanStureNielsen) [inline src/main/java/net/openhft/chronicle/queue/impl/single/StoreTailer.java:226]: This isn't part of your PR but should probably be removed as well.

File: src/test/java/net/openhft/chronicle/queue/impl/single/StoreAppenderInternalWriteBytesTest.java [hunk 5]
```diff
@@ -48,13 +47,12 @@ public void internalWriteBytesShouldBeIdempotent() throws InterruptedException {
     public void testInternalWriteBytes(int numCopiers, boolean concurrent) throws InterruptedException {
         final Path sourceDir = IOTools.createTempDirectory("sourceQueue");
         final Path destinationDir = IOTools.createTempDirectory("destinationQueue");
-
-        /**
+         /**
```
Comment (JanStureNielsen) [inline src/test/java/net/openhft/chronicle/queue/impl/single/StoreAppenderInternalWriteBytesTest.java:50]: Remove this block?

File: src/test/java/net/openhft/chronicle/queue/impl/single/StoreAppenderInternalWriteBytesTest.java [hunk 6]
```diff
@@ -158,12 +157,12 @@ public void run() {
                                 assertEquals(Long.toHexString(index), Long.toHexString(dtIndex));
                         }
                         prev.clear().append(buffer);
-//                        if (index %17 == 0) {
+//                        if (false && index %17 == 0) {
```
Comment (JanStureNielsen) [inline src/test/java/net/openhft/chronicle/queue/impl/single/StoreAppenderInternalWriteBytesTest.java:160]: Unintentional change here? Regardless, perhaps just removing is the better option, paired with the brace below.

**Review by JanStureNielsen — COMMENTED** [review]
The PR is merged already so perhaps add a polish PR...

**Pull request merged**